plugins/modules/ise_radius_integration_workflow_manager.py

```diff
@@ -734,366 +739,372 @@ def get_want_authentication_policy_server(self, auth_policy_server):
-                auth_server.update({"encryptionScheme": encryption_scheme})
-
-            if encryption_scheme == "KEYWRAP":
-                message_key = auth_policy_server.get("message_authenticator_code_key")
-                if not message_key:
-                    self.msg = "The 'message_authenticator_code_key' should not be empty if the encryption_scheme is 'KEYWRAP'."
-                    self.status = "failed"
-                    return self
-
-                message_key = str(message_key)
-                message_key_length = len(message_key)
-                if message_key_length != 20:
-                    self.msg = "The 'message_authenticator_code_key' should be exactly 20 characters."
-                    self.status = "failed"
-                    return self
-
-                auth_server.update({"messageKey": message_key})
+            if protocol is not None:
+                auth_server.update({"protocol": protocol})
+            else:
+                if not auth_server_exists:
+                    auth_server.update({"protocol": "RADIUS"})
+                else:
+                    auth_server.update({"protocol": auth_server_details.get("protocol")})
 
-                encryption_key = auth_policy_server.get("encryption_key")
-                if not encryption_key:
-                    self.msg = "The encryption_key should not be empty if encryption_scheme is 'KEYWRAP'."
-                    self.status = "failed"
-                    return self
+            auth_server.update({"port": 49})
 
-                encryption_key = str(encryption_key)
-                encryption_key_length = len(encryption_key)
-                if encryption_key_length != 16:
-                    self.msg = "The 'encryption_key' must be 16 characters long. It may contain alphanumeric and special characters."
+            if not auth_server_exists:
+                encryption_scheme = item.get("encryption_scheme")
+                if encryption_scheme not in ["KEYWRAP", "RADSEC", None]:
+                    self.msg = "The encryption_scheme should be in ['KEYWRAP', 'RADSEC']. " + \
+                               "It should not be {0}.".format(encryption_scheme)
                     self.status = "failed"
                     return self
 
-                auth_server.update({"encryptionKey": encryption_key})
+                if encryption_scheme:
+                    auth_server.update({"encryptionScheme": encryption_scheme})
 
-        if not auth_server_exists:
-            authentication_port = auth_policy_server.get("authentication_port")
-            if not authentication_port:
-                authentication_port = 1812
+                if encryption_scheme == "KEYWRAP":
+                    message_key = item.get("message_authenticator_code_key")
+                    if not message_key:
+                        self.msg = "The 'message_authenticator_code_key' should not be empty if the encryption_scheme is 'KEYWRAP'."
```

Comment:
self.msg = ("The 'message_authenticator_code_key' must not be empty when the 'encryption_scheme' is set to 'KEYWRAP'.")